hpat/tests/test_series.py

```diff
@@ -303,6 +303,32 @@ def test_impl(S):
         S = pd.Series(['aa', 'bb', 'cc'])
         pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
 
+    def test_series_astype_str_to_str_index_str(self):
```

Comment:
This is not boxing/unboxing issue, so better to revert it and do it in a separate PR.